### PR TITLE
Directory prefix and dev helper script

### PR DESCRIPTION
**Pull request by bryan-bar**
* each step given an ordering prefix
  * ex: `aws-dbt2-aurora/provision` -> `aws-dbt2-aurora/01_provision`
  * Directories now correctly ordered by filesystem when viewed and can be ordered by the prefix
* environment file created under each benchmark
  * ex: `aws-dbt2-aurora/environment`
* Helper script `run_benchmark.sh` created
  * able to exclude steps by name or index as well as start at any step by name or index
  * user prompted for input if any values within environment file are empty and will be copied beside the original for re-use
```shell
chmod +x run_benchmark.sh
# help with options
./run_benchmark.sh -h
# start a benchmark
./run_benchmark.sh -b aws-dbt2-aurora
```

### CODE REVIEW

**Review by dougortiz — APPROVED** [review]
LGTM

**Comment by vibhorkumar123** [comment]
Something doesn't look good here. Files changed 305?

**Comment by bryan-bar** [comment]
> Something doesn't look good here. Files changed 305?

@vibhorkumar123 many of the changes are just renaming of the directory, moving variables to environment file for easy view/updating and removing of unneeded or duplicate variables.

**Comment by mw2q** [comment]
> Something doesn't look good here. Files changed 305?

Yeah, this is one of those things that if we addressed earlier, then it wouldn't need this many adjustments.  :)  It's something that grows very rapidly left unchecked.

File: README.md [hunk 1]
```diff
@@ -33,18 +33,25 @@ Execution of a benchmark is accomplished by following the steps below
...
+- `03_prepare` - Prepares and sets up the benchmark execution
+- `04_execute` - Executes the benchmark
+- `05_push_results` - Upload the results from the benchmark into AWS S3 Bucket
+- `06_unprovision` - Destroys the previously provisioned infrastructure
 
 **Each directory contains a `run.sh` script file that executes the ansible playbooks required for the benchmark**
 
+** For development, you can use the run_benchmark.sh script to assist in running and debugging each step.
+```shell
+chmod +x run_benchmark.sh
```
Comment (mw2q) [inline README.md:50]: i think the file should be checked in with execute permissions. then this line is unnecessary.

File: run_benchmark.sh [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+#   * Each benchmark directory has directories with a numberic prefix ordering
+#     * Within each directory is a run.sh script to kick off that step
+#     * ex: aws-dbt2-aurora/00_validate/run.sh
+#   * Each benchmark directory has an environment file to source required variables to run the benchmark
+#     * ex: aws-dbt2-aurora/environment
+# Examples availble below in usage()
+
+usage () {
+    cat << EOF
+Usage: $0 -b benchmark_directory [-e environment_file] [-s starting_step ] [-x comma_separated_steps] [-u < true | false > ]
```
Comment (mw2q) [inline run_benchmark.sh:14]: i like using `$(basename "$0") ` :)

File: run_benchmark.sh [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+                exit 0;;
+            s)
+                start_step="${OPTARG}";;
+            x)
+                IFS=',' read -ra exclude_steps <<< "${OPTARG}";;
+            u) 
+                user_input="${OPTARG}";;
+            *)
+                usage;
+                exit 0;;
```
Comment (mw2q) [inline run_benchmark.sh:52]: this is unknown argument case right? exit should be non-zero unless i'm wrong.

File: run_benchmark.sh [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        esac
+    done
+    if (( $OPTIND == 1 )); then
+        usage
+        exit 0
```
Comment (mw2q) [inline run_benchmark.sh:57]: i'm pretty sure about this one. :) i think we need non-zero exit code for unknown options.

**Review by mw2q — CHANGES_REQUESTED** [review]
That's some pretty advanced (to me) bash stuff.  I just focused on the run_benchmark.sh script, but looks mostly good to me.

**Pull request merged**